### DISCUSSION TITLE
Filling in modbus server response

### DISCUSSION THREAD

**Discussion by fahadalisarwar1** [eModbus/eModbus#78] [eModbus applications]
How do i put the desired data in the modbus response
```
    response.add(request.getServerID(), request.getFunctionCode(), (uint8_t)(words * 2));
    // Fill response with requested data
    for (uint16_t i = address; i < address + words; ++i) {
      response.add(i);
    }
```

when i add the desired data in response.add, the master says invalid CRC.

**Comment by Miq1** [comment]
The ``add()`` does not affect the CRC calculation at all, that is done separately, after the message has been returned. I suppose your code snippet above is part of some callback function for function code 03?

What value has ``words``? Can it be too large to fit into a single message (125 words with FC 0x03)?

**Reply by fahadalisarwar1** [reply]
i am just reading two words, yes the above mentioned code is in the callback method. if i run the exact callback function as mentioned above, the data is received properly but as soon as i change the parameter of the add() method, the master says invalid CRC.

**Reply by Miq1** [reply]
Okay then: in which of the two add() calls do you change the parameters? In the first line or in the for loop? And what is the exact change that will result in the client complaining about the CRC?
